File: 17. CustomDialogboxDemo/app/src/main/java/com/dilip/customdialogboxdemo/MainActivity.java
package com.dilip.customdialogboxdemo;

import android.app.Dialog;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Create a custom dialog
        Dialog dialog = new Dialog(this);
        dialog.setContentView(R.layout.custom_dialog_layout);
        dialog.setCancelable(false); // Prevent dialog dismissal on outside touch

        // Find the "Okay" button in the custom dialog
        Button btnOkay = dialog.findViewById(R.id.btnOkay);

        // Set a click listener for the "Okay" button
        btnOkay.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Display a toast message and dismiss the dialog
                Toast.makeText(MainActivity.this, "Close", Toast.LENGTH_SHORT).show();
                dialog.dismiss();
            }
        });

        // Show the custom dialog
        dialog.show();
    }
}

File: 01. ActivityCycleDemo/app/src/main/java/com/dilip/activitycycledemo/MainActivity.kt
package com.dilip.activitycycledemo

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View.INVISIBLE
import android.view.View.VISIBLE
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        Log.i("MYTAG", "MainActivity: OnCreate")
        val greetingTextView = findViewById<TextView>(R.id.HelloWorld)
        val inputField = findViewById<EditText>(R.id.tvText)
        val submitButton = findViewById<Button>(R.id.button)
        val offersButton = findViewById<Button>(R.id.btnOffers)

        var enteredName = ""
        submitButton.setOnClickListener{
            enteredName = inputField.text.toString()
            if (enteredName == "") {
                offersButton.visibility = INVISIBLE
                greetingTextView.text = ""
                Toast.makeText(this@MainActivity, "Please, Enter your name", Toast.LENGTH_SHORT).show()
            } else {
                val message = "Welcome $enteredName"
                Log.i("MYTAG", message)
                greetingTextView.text = message
                Log.i("MYTAG", "After the displaying the text message on the TextView")
                inputField.text.clear()
                offersButton.visibility = VISIBLE
            }

            offersButton.setOnClickListener {
                val intent = Intent(this, SecondActivity::class.java)
                intent.putExtra("USER", enteredName)
                startActivity(intent)
            }

        }

    }

    override fun onStart() {
        super.onStart()
        Log.i("MYTAG", "MainActivity: OnStart")
        Toast.makeText(getApplicationContext(),"OnStart",Toast.LENGTH_SHORT).show(); 
    }

    override fun onResume() {
        super.onResume()
        Log.i("MYTAG", "MainActivity: OnResume")
        Toast.makeText(getApplicationContext(),"onResume",Toast.LENGTH_SHORT).show(); 
    }

    override fun onPause() {
        super.onPause()
        Log.i("MYTAG", "MainActivity: OnPause")
        Toast.makeText(getApplicationContext(),"onPause",Toast.LENGTH_SHORT).show(); 
    }

    override fun onStop() {
        super.onStop()
        Log.i("MYTAG", "MainActivity: OnStop")
        Toast.makeText(getApplicationContext(),"onStop",Toast.LENGTH_SHORT).show(); 
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i("MYTAG", "MainActivity: OnDestroy")
        Toast.makeText(getApplicationContext(),"onDestroy",Toast.LENGTH_SHORT).show(); 
    }

    override fun onRestart() {
        super.onRestart()
        Log.i("MYTAG", "MainActivity: OnRestart")
        Toast.makeText(getApplicationContext(),"onRestart",Toast.LENGTH_SHORT).show(); 
    }

}